src/bytes.rs

```diff
@@ -0,0 +1,197 @@
+use std::{convert::TryInto, str::FromStr};
+
+use num_bigint::BigUint;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::quote;
+use syn::{Error, LitInt};
+
+pub fn bytes(input: TokenStream2) -> TokenStream2 {
+    let lit = match syn::parse2::<LitInt>(input) {
+        Ok(lit) => lit,
+        Err(e) => return e.to_compile_error(),
+    };
+
+    // Get the raw integer literal as it appears in the token stream.
+    let raw = lit.to_string();
+
+    // Remove leading negative sign, and any underscores between digits.
+    let normalized = raw.replace(['-', '_'], "");
```

Comment:
I think this should error on negative.

Reply:
The macro already errors on negative, but it did so near the end so I was removing the negative here to avoid it when counting zeros.

But, this approach makes it look like negatives are supported when they aren't. They aren't supported because the BigUint parse later on.

In 9f9fe48 I've moved that parse earlier so we take out the need to deal with negatives when counting the zeros. It makes the function clearer I think too.

In that commit I also made the error displayed negative more explicit and made it a proper compile error.